doc/internal/consensus/Consensus-Protocol.md

```diff
@@ -0,0 +1,127 @@
+# The consensus
+
+Before talking about consensus, we need to discuss some logistics based on how the systems can co-exist.
+
+* Communication: Distributed systems need a method to communicate between each other. Remote Procedure Calls is the mechanism using which a standalone server can talk to another. The existing `network` layer of the database will handle all the communication between servers. 
```

Comment:
However, only the communication. Messages etc. need to be specified and are part of the Raft implementation.